File: 프로그래머스/2/42578. 의상/의상.py
def solution(clothes):
    dict = {}
    for name, type in clothes:
        if type in dict:
            dict[type] += 1
        else:
            dict[type] = 1
            
    answer = 1  
    for count in dict.values():
         answer*= count + 1
    return answer -1

File: 프로그래머스/2/42578. 의상/의상.java
import java.util.HashMap;
import java.util.Map;

class Solution {
    public int solution(String[][] clothes) {
        int answer = 1;

        Map<String, Integer> map = new HashMap<>();
        for (String[] clothe : clothes) {
            map.put(clothe[1], map.getOrDefault(clothe[1], 0) + 1);
        }

        for (int value : map.values()) {
            System.out.println(value);
            answer *= value + 1;
        }

        return answer - 1;
    }
}
